extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -218,3 +322,72 @@ async function findDirWithFile(dir: string, ...toFind: string[]): Promise<string
   }
   return;
 }
+
+function looksLikeLgtmUrl(lgtmUrl: string | undefined): lgtmUrl is string {
+  if (!lgtmUrl) {
+    return false;
+  }
+
+  try {
+    const uri = Uri.parse(lgtmUrl, true);
+    if (uri.scheme !== "https") {
+      return false;
+    }
+
+    if (uri.authority !== "lgtm.com" && uri.authority !== "www.lgtm.com") {
+      return false;
+    }
+
+    const paths = uri.path.split("/").filter((segment) => segment);
+    return paths.length === 4 && paths[0] === "projects" && paths[1] === "g";
+  } catch (e) {
+    return false;
+  }
+}
+
+async function convertToDatabaseUrl(lgtmUrl: string) {
+  try {
+    const uri = Uri.parse(lgtmUrl, true);
+    const paths = ["api", "v1.0"].concat(
+      uri.path.split("/").filter((segment) => segment)
+    );
+    const projectUrl = `https://lgtm.com/${paths.join("/")}`;
+    const projectResponse = await fetch(projectUrl);
+    const projectJson = await projectResponse.json();
+
+    if (projectJson.code === 404) {
+      throw new Error();
+    }
+
+    const language = await promptForLanguage(projectJson);
+    if (!language) {
+      return;
+    }
+    return `https://lgtm.com/${[
+      "api",
+      "v1.0",
+      "snapshots",
+      projectJson.id,
+      language,
+    ].join("/")}`;
+  } catch (e) {
+    throw new Error(`Invalid LGTM URL: ${lgtmUrl}`);
```

Comment:
You should probably include the error message that LGTM returns on error. If a user provides a valid LGTM URL but the wrong language, we would wrongly complain that the LGTM URL is invalid.

Reply:
Sure. I can do something, but just note that a user can't request an invalid language since we only allow a user to select from the languages exposed in the project metadata.

Reply:
That's true. Although more precise error message are likely to be helpful in case we ever need to debug a failure.

Reply:
Sure. I'll send these messages to the log files.